sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/connections/workspace_connection.py

```diff
@@ -422,15 +405,24 @@ def _get_schema_class_from_type(cls, conn_type: Optional[str]) -> Type:
         :rtype: Type
         """
         if conn_type is None:
-            return WorkspaceConnection
+            return WorkspaceConnectionSchema
+        entity_class = cls._get_entity_class_from_type(conn_type)
+        return entity_class._get_schema_class()
 
-        cat = camel_to_snake(conn_type).lower()
-        conn_class: Type = WorkspaceConnectionSchema
-        if cat == camel_to_snake(ConnectionCategory.AZURE_OPEN_AI).lower():
-            conn_class = OpenAIWorkspaceConnectionSchema
-        elif cat == camel_to_snake(ConnectionCategory.COGNITIVE_SEARCH).lower():
-            conn_class = AzureAISearchWorkspaceConnectionSchema
-        elif cat == camel_to_snake(ConnectionCategory.COGNITIVE_SERVICE).lower():
-            conn_class = AzureAIServiceWorkspaceConnectionSchema
+    @classmethod
+    def _get_required_metadata_fields(cls) -> List[str]:
+        """Helper function that returns the required metadata fields for specific workspace
```

Comment:
Does this help? https://docs.python.org/3/library/abc.html#abc.abstractclassmethod

Reply:
Oh! That's a new annotation to me, thanks! That said I don't think I want to apply it here, since I do still want the main class to be callable (as opposed to throwing errors or warnings).